src/actions.rs

```diff
@@ -185,15 +200,16 @@ impl ActionHandler {
         let analysis = self.analysis.clone();
 
         let rustw_handle = thread::spawn(move || {
-            let file_name = doc.textDocument.file_name();
+            let file_name = parse_file_path(&doc.text_document.uri).unwrap();
             let symbols = analysis.symbols(&file_name).unwrap_or(vec![]);
             t.unpark();
 
             symbols.into_iter().map(|s| {
                 SymbolInformation {
                     name: s.name,
-                    kind: VscodeKind::from(s.kind) as u32,
-                    location: Location::from_span(&s.span),
+                    kind: source_kind_from_def_kind(s.kind),
+                    location: ls_util::location_from_span(&s.span),
+                    container_name: None // TODO: more info could be added here
```

Comment:
nit: prefer FIXME over TODO for things being left in the code (i.e., no TODOs should get committed)